### PR TITLE
Add Support for Supreme Ego

### LINKED ISSUE #4290: Supreme Ego effect not properly calculated
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### What is the expected behaviour/value?

Supreme ego's new effect is to add 1% _more_ aura effect per 2% of mana reserved. This means that, say, Hatred at 50% reservation (70% after Ego) would have 37.5% more aura effect, which PoB seems to completely ignore. I have tested the keystone in game and the damage of my main skill is visibly and numerically affected. 

### What is the actual behaviour/value?

PoB ignores the aura effect multiplier that should be added with the keystone. In my character I use Hatred aura, with 50% base reservation and 75% with the keystone. This should add 37.5% more effect, as is shown in-game before and after picking the keystone:
image
image
PoB ignores it as shown below:
image

### How to reproduce the issue

Simply run an aura of your choice (in my example I use Hatred as that is my main aura) and pick/remove the keystone in-game and then in PoB and you will see the damage in PoB is unchanged and you have to add the "more aura effect" by hand under Custom Modifiers tab.

### Build code

```shell
https://pastebin.com/EvwkYsAw
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by QuickStick123**
Fixes #4290.

### Description of the problem being solved:
Add support for Supreme Ego and disable auras if they reserve more than maximum mana.
Greater than 100% reduced reservation and essence worm will behave weirdly with this patch so #4514 should be merged first.

### Steps taken to verify a working solution:
- Checking various auras ingame it seems any skill that have the aura tag will be effected by the mod this means tempest shield, blood and sand aren't affected at all.
- Any skill of your that apply any aura to anything but you will not work, blasphemy, pride, smite aura, etc.
- I haven't tested the mine auras but I presume they won't work. The current code disables them as well the aura effect only is based on one mine which I am not sure is correct but they are disabled with supreme ego anyway so it doesn't matter.
- A curse in profane proxy won't work either but I don't think we fully support profane proxy so this doesn't work in pob.
- I have the mana reserved multiplier capped at 100% (50% more aura effect) and I skip adding the aura if its reservation exceed maximum of resource. This should prevent easy abuse cases like slapping on a essence worm to try and get aura effect as it will cause those aura that exceed 100% to not be enabled.
- The discrepancy for discipline in the screen shot is caused by low mana rounding errors I think.

### Link to a build that showcases this PR:
https://poe.ninja/pob/BY5

image
### Before screenshot:
image
image

### After screenshot:
image
image

### CODE REVIEW

**Review by Wires77 — COMMENTED** [review]
One missing edge case here is if you have a blasphemy curse + a self-cast curse, the self-cast one doesn't get registered, since we're at the curse limit because of the other curse already.

**Comment by QuickStick123** [comment]
Fixed curse issue for both exceeding reservation and supreme ego disable.

File: src/Classes/ModStore.lua [hunk 1]
```diff
@@ -219,6 +219,18 @@ function ModStoreClass:GetMultiplier(var, cfg, noMod)
 end
 
 function ModStoreClass:GetStat(stat, cfg)
+	if stat == "ManaReservedPercent" then
+		local manaBase = 0
```
Comment (Wires77) [inline src/Classes/ModStore.lua:223]: Unnecessary variable when outside the loop
```suggestion
```

File: src/Classes/ModStore.lua [hunk 2]
```diff
@@ -219,6 +219,18 @@ function ModStoreClass:GetMultiplier(var, cfg, noMod)
...
+		local reservedPercentMana = 0
+		for _, activeSkill in ipairs(self.actor.activeSkillList) do
+			if (activeSkill.skillTypes[SkillType.Aura] and not activeSkill.skillFlags.disable and activeSkill.buffList and activeSkill.buffList[1] and activeSkill.buffList[1].name == cfg.skillName) then
+				manaBase = activeSkill.skillData["ManaReservedBase"] and activeSkill.skillData["ManaReservedBase"] or 0
+				reservedPercentMana =  manaBase / self.actor.output["Mana"] * 100
```
Comment (Wires77) [inline src/Classes/ModStore.lua:228]: Simplification
```suggestion
				local manaBase = activeSkill.skillData["ManaReservedBase"] or 0
				reservedPercentMana = manaBase / self.actor.output["Mana"] * 100
```

File: src/Modules/ModParser.lua [hunk 3]
```diff
@@ -1054,6 +1055,7 @@ local modTagList = {
 	["per allocated mastery passive skill"] = { tag = { type = "Multiplier", var = "AllocatedMastery" } },
 	["per allocated notable passive skill"] = { tag = { type = "Multiplier", var = "AllocatedNotable" } },
 	-- Per stat
+	["per (%d+)%% of maximum mana they reserve"] = function(num) return { tag = { type = "PerStat", stat =  "ManaReservedPercent", div = num } } end,
```
Comment (Wires77) [inline src/Modules/ModParser.lua:1058]:
```suggestion
	["per (%d+)%% of maximum mana they reserve"] = function(num) return { tag = { type = "PerStat", stat = "ManaReservedPercent", div = num } } end,
```

**Review by Wires77 — APPROVED** [review]
Found some nitpicks, but I'll just push those changes myself, this looks good otherwise

**Pull request merged**